src/compiler/utilities.ts

```diff
@@ -1884,8 +1884,8 @@ namespace ts {
             writeTextOfNode,
             writeLiteral,
             writeLine,
-            increaseIndent: () => indent++,
-            decreaseIndent: () => indent--,
+            increaseIndent: () => { indent++; },

```

Comment:
No, I think the rule itself is effectively that you have to separate the mutation of your variable from reading its (potentially new) value. If people find this too restrictive, we can always change it back.